octopoes/octopoes/models/path.py

```diff
@@ -61,7 +61,10 @@ def reverse(self) -> Segment:
             self.source_type,
         )
 
-    def __eq__(self, other: Segment) -> bool:
+    def __eq__(self, other: object) -> bool:
+        if not isinstance(other, Segment):
+            return NotImplemented
```

Comment:
You probably meant `raise` instead of `return`?

Reply:
No, see https://docs.python.org/3/reference/datamodel.html#object.__eq__